### ISSUE TITLE
Pivot fields display blank when editing BelongsToMany relationships with pivot accessor

### ISSUE THREAD

**Issue by chrispage1** [laravel/nova-issues#2650]
- Laravel Version: 7.13.0
- Nova Version: 3.6.0
- PHP Version: 7.4.6
- Database Driver & Version: MariaDB 10.3

### Description:

When using a custom BelongsToMany pivot accessor string, you are unable to update pivot fields. This issue has been raised before but is still not resolved (see #1694)

### Steps To Reproduce:

Model (Team is exactly the same in reverse). Notice I am using `->as('profileTeam')`. This is the custom accessor:

```
class Profile extends Model
{
    /**
     * Retrieve a list of all teams this profile
     * has ever been associated to.
     *
     * @return BelongsToMany
     */
    public function teams(): BelongsToMany
    {
        return $this->belongsToMany(Team::class)
            ->using(ProfileTeam::class)
            ->withPivot(['title', 'started_at', 'ended_at'])
            ->as('profileTeam');
    }
}
```

Nova resource, (again, there is the inverse on the Team resource):

```
class Profile extends Resource
{
    ...

    /**
     * Get the fields displayed by the resource.
     *
     * @param  \Illuminate\Http\Request  $request
     * @return array
     */
    public function fields(Request $request)
    {
        return [
            BelongsToMany::make('Teams')
                ->fields(function () {
                    return [
                        Text::make('Title')
                            ->onlyOnForms()
                            ->help('Optional title / position'),

                        Date::make('Start Date', 'started_at')
                            ->rules(['required'])
                            ->onlyOnForms(),

                        Date::make('End Date', 'ended_at')
                            ->onlyOnForms()
                            ->nullable()
                    ];
                })
        ];
    }
```

When creating the relationship, I can populate the title, start date and end date fields without a problem. It's when I go to edit them that all of the fields appear blank. This is caused by the `UpdatePivotFieldController` as mentioned by @alshf89 back in February -https://github.com/laravel/nova-issues/issues/1694#issuecomment-591950410

Current method within `UpdatePivotFieldController` is - 

```
public function index(NovaRequest $request)
    {
        $model = $request->findModelOrFail();

        $model->setRelation(
            $model->{$request->viaRelationship}()->getPivotAccessor(),
            $model->{$request->viaRelationship}()->withoutGlobalScopes()->findOrFail($request->relatedResourceId)->pivot
        );

        return response()->json(
            $request->newResourceWith($model)->updatePivotFields(
                $request,
                $request->relatedResource
            )->all()
        );
    }
```

Notice that the `pivot` is hard coded. It should be - 

```
public function index(NovaRequest $request)
    {
        $model = $request->findModelOrFail();

        $accessor = $model->{$request->viaRelationship}()->getPivotAccessor();

        $model->setRelation(
            $accessor,
            $model->{$request->viaRelationship}()->withoutGlobalScopes()->findOrFail($request->relatedResourceId)->{$accessor}
        );

        return response()->json(
            $request->newResourceWith($model)->updatePivotFields(
                $request,
                $request->relatedResource
            )->all()
        );
    }
```

**Comment by alshf89** [comment]
I can send a PR 😬

**Comment by chrispage1** [comment]
> I can send a PR 😬

@alshf89 I've just submitted a PR fingers crossed!

**Comment by alshf89** [comment]
So lets see

**Comment by alshf89** [comment]
@chrispage1 it fixes in Nova new release `3.7.0`

**Comment by chrispage1** [comment]
@alshf89 - thanks - the PR got approved and has been merged in :)
